tests/phpunit/CRM/Utils/GeocodeTest.php

```diff
@@ -42,18 +49,33 @@ public function testGeocodeMethodOff() {
     $result = civicrm_api3('Contact', 'create', $params);
     $contact_values = array_pop($result['values']);
     $address_values = array_pop($contact_values['api.Address.create']['values']);
-    // We should get a geo code setting.
-    $this->assertApproxEquals('38.89', CRM_Utils_Array::value('geo_code_1', $address_values), 1);
 
-    // Set geocodeMethod to empty.
-    $config = CRM_Core_Config::singleton();
-    $config->geocodeMethod = '';
+    $this->assertArrayNotHasKey('geo_code_1', $address_values, 'No geocoding when geocodeMethod is empty');
+
+    // Run the geocode job on that specific contact
+    $config->geocodeMethod = 'CRM_Utils_Geocode_Google';
 
-    // Do it again. This time, we should not geocode.
-    $new_result = civicrm_api3('Contact', 'create', $params);
-    $new_contact_values = array_pop($new_result['values']);
-    $new_address_values = array_pop($new_contact_values['api.Address.create']['values']);
-    $this->assertArrayNotHasKey('geo_code_1', $new_address_values, 'No geocoding when geocodeMethod is empty');
+    try {
+      $params_geocode = array(
+        'start' => $contact_values['id'],
+        'end' => $contact_values['id'] + 1,
+        'geocoding' => 1,
+        'parse' => 0,
+      );
+      $result_geocode = civicrm_api3('Job', 'geocode', $params_geocode);
+    }
+    catch (CiviCRM_API3_Exception $e) {
+      if ($e->getMessage() == 'A fatal error was triggered: Aborting batch geocoding. Hit the over query limit on geocoder.') {
+        $this->markTestIncomplete('Job.geocode error_message: A fatal error was triggered: Aborting batch geocoding. Hit the over query limit on geocoder.');
+      }
+    }
```

Comment:
@DanielvV If the exception message is *not* about the anticipated query limit, should we re-throw the exception? Intuitively, it feels like the is only supposed to suppress a specific error -- any other errors should still be errors.

Breaking it down, it could be argued -- eg if `Job.geocode` failed, then it would continue to execute lines 72-78, which would then fail, so we'd still end up at an appropriate result (failed test). However, it's probably easier to understand the problem if you rethrow the original exception (because then the error report will be more precise/accurate).

Reply:
Makes perfect sense